tests/post/steps/test_salt_metalk8s_monitoring.py

```diff
@@ -0,0 +1,219 @@
+        raise ValueError(f"Invalid alert state: {alert_state}")
+    if alert_state != "all":
+        command.append(f"state={alert_state}")
+    command.append("--out json")
+    command.append("--log-level quiet")
+    result = run_salt_command(
+        host,
+        command,
+        ssh_config,
+    )
+    context[f"alerts_{alert_state}"] = json.loads(result.stdout)
+
+
+# }}}
+# Then {{{
+
+
+@then(
+    parsers.cfparse(
+        "there should{negated:Negation?} be silences '{silence_state}'",
+        extra_types={"Negation": negation},
+    )
+)
+def confirm_silences(context, negated, silence_state):
+    result = bool(context[f"silences_{silence_state}"])
+    if negated:
+        assert not result, f"Silences in {silence_state} state found"
+    else:
+        assert result, f"Silences in {silence_state} state not found"
+
+
+@then(
+    parsers.cfparse(
+        "we should{negated:Negation?} find the silence for the '{alert_name}' alert "
+        "in the '{silence_state}' silences list",
+        extra_types={"Negation": negation},
+    ),
+)
+def find_silence_for_alert(context, negated, alert_name, silence_state):
+    try:
+        silence = next(
+            silence
+            for silence in context[f"silences_{silence_state}"]
+            if silence["matchers"][0]["value"] == alert_name
+        )
+    except StopIteration:
+        silence = None
+    if negated:
+        assert not silence, f"Silence for {alert_name} alert found"
+        context.pop(f"silence_{alert_name}", None)
+    else:
+        assert silence, f"Silence for {alert_name} alert not found"
+        context[f"silence_{alert_name}"] = silence
+
+
+@then(
+    parsers.cfparse(
+        "the '{alert_name}' alert should{negated:Negation?} be "
+        "in the list of '{alert_state}' alerts",
+        extra_types={"Negation": negation},
+    ),
+)
+def confirm_alert(context, alert_name, negated, alert_state):
+    result = any(
+        alert["labels"]["alertname"] == alert_name
+        for alert in context[f"alerts_{alert_state}"]
+    )
+    if negated:
+        assert not result, f"{alert_name} alert is in {alert_state} alerts list"
+    else:
+        assert result, f"{alert_name} alert is not in {alert_state} alerts list"
+
+
+@then(
+    parsers.parse(
+        "we poll the '{alert_state}' alerts "
+        "until find the '{alert_name}' alert or "
+        "fail after {timeout:d} seconds"
+    ),
+)
+def poll_alerts(host, ssh_config, context, alert_state, alert_name, timeout):
+    start = time.time()
+    while time.time() - start < int(timeout):
+        call_metalk8s_monitoring_get_alerts(host, ssh_config, context, alert_state)
+        result = any(
+            alert["labels"]["alertname"] == alert_name
+            for alert in context[f"alerts_{alert_state}"]
+        )
+        if result:
+            return
+        time.sleep(10)
+    assert False, f"{alert_name} alert not found in {alert_state} alerts list"
```

Comment:
You should use `utils.retry` it's what we used everywhere to do such retry

Reply:
Done.